### PR TITLE
feat: add ops.SimpleCase to translate and decompile

### PR DESCRIPTION
**Pull request by gforsyth**

### CODE REVIEW

File: ibis_substrait/compiler/translate.py [hunk 1]
```diff
@@ -906,3 +906,25 @@ def aggregation(
...
+    expr: ir.TableExpr,
+    compiler: SubstraitCompiler,
+    **kwargs: Any,
+) -> stalg.Expression:
+    # the field names for an `if_then` are `if` and `else` which means we need
+    # to pass those args in as a dictionary to not run afoul of SyntaxErrors`
+    _ifs = []
+    for case, result in zip(op.cases, op.results):
+        _if = {
+            "if": translate(op.base == case, compiler),
```
Comment (cpcloud) [inline ibis_substrait/compiler/translate.py:923]: Probably wanna thread `**kwargs` thought the `translate` calls.

**Review by cpcloud — APPROVED** [review]
Small comment about threading `**kwargs`, otherwise LGTM.

File: ibis_substrait/compiler/translate.py (hunk 1 above)
Comment (gforsyth) [reply ibis_substrait/compiler/translate.py]: :+1: done

**Comment by gforsyth** [comment]
xref: https://github.com/ibis-project/tpc-queries/pull/3

**Pull request merged**